sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs

```diff
@@ -36,36 +36,64 @@ public abstract class AuthenticationEventRequest<TResponse, TData> : Authenticat
 
         /// <summary>Validates the response and creates the IActionResult with the json payload based on the status of the request.</summary>
         /// <returns>IActionResult based on the EventStatus (UnauthorizedResult, BadRequestObjectResult or JsonResult).</returns>
-        public async override Task<AuthenticationEventResponse> Completed()
+        public override AuthenticationEventResponse Completed()
         {
             try
             {
                 if (RequestStatus == RequestStatusType.Failed || RequestStatus == RequestStatusType.ValidationError)
                 {
-                    Response.MarkAsFailed(new Exception(string.IsNullOrEmpty(StatusMessage) ? AuthenticationEventResource.Ex_Gen_Failure : StatusMessage), true);
+                    return Failed(new AuthenticationEventTriggerRequestValidationException(string.IsNullOrEmpty(StatusMessage) ? AuthenticationEventResource.Ex_Gen_Failure : StatusMessage));
                 }
                 else if (RequestStatus == RequestStatusType.TokenInvalid)
                 {
-                    Response.MarkAsUnauthorized();
+                    Unauthorized();
                 }
             }
             catch (Exception ex)
             {
-                return await Failed(ex, true).ConfigureAwait(false);
+                return Failed(new AuthenticationEventTriggerRequestValidationException(ex.Message));
```

Comment:
I don't think this should be a validation exception

Reply:
Updated to just Exception